pathcore/network_permutation_test.py

```diff
@@ -0,0 +1,185 @@
+"""Methods for permuting the pathways across constructed features of one
+or more unsupervised models.
+A permutation maintains the number of unique pathways that correspond to a
+feature. If applicable, it also preserves the side (positive or negative)
+in which pathway was overrepresented as well.
+"""
+import csv
+
+from statsmodels.sandbox.stats.multicomp import multipletests
+
+
+def aggregate_permuted_network(observed_networks):
+    """This method handles the case where multiple observed networks are
+    generated (e.g. from models produced by different random seed
+    initializations of an unsupervised feature construction algorithm).
+    We handle analysis of multiple networks by aggregating them; likewise,
+    we require that the permutation test generates N aggregate permuted
+    networks.
+
+    Parameters
+    -----------
+    observed_networks : list(CoNetwork)
+      the list of observed networks, generated from models produced by
+      different random seed initializations of the same unsupervised feature
+      construction algorithm
+
+    Returns
+    -----------
+    CoNetwork, the aggregate permuted network created by generating
+      a permutation for each individual observed network and then aggregating
+      them into a single network
+    """
+    aggregate_permuted_network = None
+    for pathcore_network in observed_networks:
+        permuted_network = pathcore_network.permute_pathways_across_features()
+        if not aggregate_permuted_network:
+            aggregate_permuted_network = permuted_network
+        else:
+            aggregate_permuted_network.aggregate(permuted_network)
+    return aggregate_permuted_network
+
+
+def network_edges_permutation_test(observed_network,
+                                   permuted_networks,
```

Comment:
Hmm.. so in the PathCORE-analysis repository [that will hopefully exist soon], I have a script that shows how `network_edges_permutation_test` is used. 
In the next commit on this PR
- I'll point to that in the README
- I'll also make it clear in this function's documentation that `permuted_networks` is generated by calling the [`CoNetwork`'s `permute` function](https://github.com/greenelab/PathCORE/pull/1/files/35adf7eb0700681ceff49d2ba819191b77807f8e#diff-1ed75b07a3d5795a002b33a5fb4c1337R189) _N_ times and collecting the output in a list.

Would that make things more clear?